### ISSUE TITLE
ST11 false positive iff table is qualified and outer join is used

### ISSUE THREAD

**Issue by sbrandtb** [sqlfluff/sqlfluff#6496]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

ST11 is raised on following query:

```sql
SELECT
    t1.col1,
    t2.col2
FROM
    db1.t1
LEFT JOIN
    t2
    ON
        t1.id = t2.id
```

> Joined table 'db1.t1' not referenced elsewhere in query

Both removing the qualification (`db1.`) or changing to an `INNER JOIN` or `CROSS JOIN` makes the issue go away. There could obviously be more combinations that do not work, but I hope that they have a common root cause that can be found using those examples.

### Expected Behaviour

ST11 not raised.

### Observed Behaviour

ST11 raised.

### How to reproduce

1. create test.sql with contents from above
2. `sqlfluff lint test.sql --dialect ansi` 

### Dialect

Does not seem to matter. I tried ansi, trino, postgres

### Version

3.3.0

### Configuration

Nothing. Virgin Docker container.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
This is totally a bug. Thanks for the report. Should be an easy one to fix. I'll take this.

**Comment by fatemetardasti96** [comment]
It seems this is not fixed yet, I still have the same issue
